library/tiqr/tests/OcraTest.php

```diff
@@ -65,4 +65,38 @@ public function testChallengeResponseWithSession()
         $this->assertEquals("675831", $result);
     }
 
+    public function testTruncate() {
+        // Test RFC4226 truncate algorithm
+        // Use SHA-1 with an integer value to generate the strings to truncate
+        $tests=array(
+//          sha => expected result
+            6 => 37787,
+            23 => 24262,
+            114 => 5582,
+            203 => 3164,
+            480 => 385,
+            1784 => 313,
+            5001 => 65,
+            8012 => 12,
+            9536 => 41702,
+            22264 => 16058,
+            35619 => 3,
+            71453 => 5,
+            50979 => 16854,
+            68359 => 54266,
+            113274 => 7068,
+        );
```

Comment:
Thanks for being very thorough in your testing. Having multiple expected outcomes is useful indeed!

PHPUnit comes with a very useful feature to streamline the process of testing multiple expected outcomes. The use of test [data providers](https://phpunit.readthedocs.io/en/9.5/writing-tests-for-phpunit.html#data-providers) can help in this regard. It also gives you an option to add documentation to every test case.

For now this works great as-is!